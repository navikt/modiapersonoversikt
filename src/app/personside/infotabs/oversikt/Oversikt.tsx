import * as React from 'react';
import styled from 'styled-components';
import Oversiktskomponent from './Oversiktskomponent';
import theme, { pxToRem } from '../../../../styles/personOversiktTheme';
import VarselOversikt from './VarselOversikt';
import OppfolgingOversikt from './OppfolgingOversikt';
import YtelserOversikt from './YtelserOversikt';
import UtbetalingerOversikt from './UtbetalingerOversikt';
import MeldingerOversikt from './MeldingerOversikt';
import { INFOTABS } from '../InfoTabEnum';
import SakerOversikt from './SakerOversikt';
import ErrorBoundary from '../../../../components/ErrorBoundary';

const Style = styled.article`
    @media (${theme.media.wideScreen}) {
        display: flex;
        justify-content: center;
    }
    padding: ${theme.margin.layout};
`;

const KolonneStyle = styled.div`
    > * {
        margin-bottom: ${theme.margin.layout};
    }
    @media (${theme.media.wideScreen}) {
        &:not(:last-child) {
            margin-right: ${theme.margin.layout};
        }
    }
    flex-basis: 50%;
    max-width: ${pxToRem(800)};
`;

function Oversikt() {
    return (
        <ErrorBoundary boundaryName="Oversikt">
            <Style>
                <KolonneStyle>
                    <Oversiktskomponent
                        infotabPath={INFOTABS.UTBETALING}
                        tittel={'Utbetalinger'}
                        component={UtbetalingerOversikt}
                        hurtigtast={'U'}
                    />
                    <Oversiktskomponent
                        infotabPath={INFOTABS.OPPFOLGING}
                        tittel={'OppfÃ¸lging'}
                        component={OppfolgingOversikt}
                        hurtigtast={'O'}
                    />
                    <Oversiktskomponent
                        infotabPath={INFOTABS.VARSEL}
                        tittel={'Varsler'}
                        component={VarselOversikt}
                        hurtigtast={'V'}
                    />
                </KolonneStyle>
                <KolonneStyle>
                    <Oversiktskomponent
                        infotabPath={INFOTABS.MELDINGER}
                        tittel={'Meldinger'}
                        component={MeldingerOversikt}
                        hurtigtast={'M'}
                    />
                    <Oversiktskomponent
                        infotabPath={INFOTABS.YTELSER}
                        tittel={'Ytelser'}
                        component={YtelserOversikt}
                        hurtigtast={'Y'}
                    />
                    <Oversiktskomponent
                        tittel={'Saker'}
                        infotabPath={INFOTABS.SAKER}
                        component={SakerOversikt}
                        hurtigtast={'S'}
                    />
                </KolonneStyle>
            </Style>
        </ErrorBoundary>
    );
}

export default Oversikt;
