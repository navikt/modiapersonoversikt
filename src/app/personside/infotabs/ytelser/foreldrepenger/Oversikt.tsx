import * as React from 'react';
import { Foreldrepengerettighet } from '../../../../../models/ytelse/foreldrepenger';
import { utledFraDatoForRettighet, utledMaksDato } from './foreldrePengerUtils';
import { datoVerbose } from '../../utbetalinger/utils/utbetalingerUtils';
import DescriptionList from '../felles-styling/DescriptionList';
import YtelserBullet from '../felles-styling/YtelserBullet';
import { OversiktStyling } from '../felles-styling/CommonStylingYtelser';

interface Props {
    foreldrePenger: Foreldrepengerettighet;
}

function Oversikt({foreldrePenger}: Props) {

    const foreldrePengeRetten = {
        Foreldrepengetype: foreldrePenger.foreldrepengetype,
        Dekningsgrad: foreldrePenger.dekningsgrad + '%',
        'Rettighet fra dato': datoVerbose(utledFraDatoForRettighet(foreldrePenger)).sammensatt,
        Restdager: foreldrePenger.restDager,
        Maksdato: utledMaksDato(foreldrePenger),
        Arbeidskategori: foreldrePenger.arbeidskategori
    };

    const barnet = {
        Termindato: foreldrePenger.rettighetFom && datoVerbose(foreldrePenger.rettighetFom).sammensatt,
        Fødselsdato: foreldrePenger.barnetsFødselsdato && datoVerbose(foreldrePenger.barnetsFødselsdato).sammensatt,
        'Annen forelder': foreldrePenger.andreForeldersFnr,
        Omsorgsovertakelse: 'Ikke implementert',
        'Foreldre av samme kjønn': 'Ikke implementert',
        'Antall barn': foreldrePenger.antallBarn
    };

    return (
        <OversiktStyling>
            <YtelserBullet tittel="Om foreldrepengeretten">
                <DescriptionList entries={foreldrePengeRetten}/>
            </YtelserBullet>
            <YtelserBullet tittel="Om barnet">
                <DescriptionList entries={barnet}/>
            </YtelserBullet>
        </OversiktStyling>
    );
}

export default Oversikt;
