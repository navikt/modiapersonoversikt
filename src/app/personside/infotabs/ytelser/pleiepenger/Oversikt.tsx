import * as React from 'react';
import { Pleiepengerettighet } from '../../../../../models/ytelse/pleiepenger';
import YtelserInfoGruppe from '../felles-styling/YtelserInfoGruppe';
import DescriptionList from '../../../../../components/DescriptionList';
import { getSisteVedtakForPleiepengerettighet } from './pleiepengerUtils';
import { formaterDato } from '../../../../../utils/stringFormatting';
import { utledKjønnFraFødselsnummer } from '../../../../../utils/fnr-utils';
import { Kjønn } from '../../../../../models/person/person';
import styled from 'styled-components/macro';
import ArbeidsForholdListe from './arbeidsforhold/ArbeidsforholdListe';
import theme from '../../../../../styles/personOversiktTheme';

interface Props {
    pleiepenger: Pleiepengerettighet;
}

const OversiktStyling = styled.div`
    padding: ${theme.margin.layout};
    display: flex;
    flex-wrap: wrap;
    > * {
        flex-basis: 40%;
        flex-grow: 1;
    }
`;

function getKjønnString(fnr: string): string {
    switch (utledKjønnFraFødselsnummer(fnr)) {
        case Kjønn.Mann:
            return 'gutt';
        case Kjønn.Kvinne:
            return 'jente';
        case Kjønn.Diskresjonskode:
            return 'diskresjonskode';
        default:
            return '';
    }
}

function Oversikt({ pleiepenger }: Props) {
    const gjeldeneVedtak = getSisteVedtakForPleiepengerettighet(pleiepenger);
    const kjønn = getKjønnString(pleiepenger.barnet);

    const omPleiepengerettenEntries = {
        'Fra og med': gjeldeneVedtak ? formaterDato(gjeldeneVedtak.periode.fom) : '',
        'Til og med': gjeldeneVedtak ? formaterDato(gjeldeneVedtak.periode.tom) : '',
        Pleiepengegrad: gjeldeneVedtak ? gjeldeneVedtak.pleiepengegrad + '%' : '',
        ['Barnet (' + kjønn + ')']: pleiepenger.barnet,
        'Annen forelder': pleiepenger.andreOmsorgsperson
    };

    return (
        <OversiktStyling>
            <YtelserInfoGruppe tittel="Om pleiepengeretten">
                <DescriptionList entries={omPleiepengerettenEntries} />
            </YtelserInfoGruppe>
            <YtelserInfoGruppe tittel="Arbeidssituasjon">
                <ArbeidsForholdListe pleiepengerettighet={pleiepenger} />
            </YtelserInfoGruppe>
        </OversiktStyling>
    );
}

export default Oversikt;
