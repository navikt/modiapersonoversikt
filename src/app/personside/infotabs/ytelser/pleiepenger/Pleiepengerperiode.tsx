import * as React from 'react';
import { Pleiepengeperiode } from '../../../../../models/ytelse/pleiepenger';
import styled from 'styled-components/macro';
import YtelserPeriode from '../felles-styling/YtelserPeriode';
import { Ingress } from 'nav-frontend-typografi';
import { formaterDato, NOKellerNull, prosentEllerNull } from '../../../../../utils/stringFormatting';
import theme from '../../../../../styles/personOversiktTheme';
import { StyledTable } from '../../../../../utils/table/StyledTable';

interface Props {
    periode: Pleiepengeperiode;
    periodeNummer: number;
}

const VedtaksTable = styled.div`
    ${theme.gråttPanel};
    margin: ${theme.margin.layout};
    > *:not(:last-child) {
        margin-bottom: 1rem;
    }
`;

function Pleiepengerperiode({ periode, ...props }: Props) {
    const tittelRekke = ['Fra og med', 'Til og med', 'Bruttobeløp', 'Anvist utbetaling', 'Dagsats', 'Pleiepengegrad'];
    const rows = periode.vedtak.map(vedtak => [
        formaterDato(vedtak.periode.fom),
        formaterDato(vedtak.periode.tom),
        NOKellerNull(vedtak.bruttobeløp),
        formaterDato(vedtak.anvistUtbetaling),
        NOKellerNull(vedtak.dagsats),
        prosentEllerNull(vedtak.pleiepengegrad)
    ]);

    return (
        <YtelserPeriode tittel={`Periode ${props.periodeNummer} - ${formaterDato(periode.fom)}`}>
            <VedtaksTable>
                <Ingress>Anviste utbetalinger</Ingress>
                <StyledTable tittelRekke={tittelRekke} rows={rows} />
            </VedtaksTable>
        </YtelserPeriode>
    );
}

export default Pleiepengerperiode;
