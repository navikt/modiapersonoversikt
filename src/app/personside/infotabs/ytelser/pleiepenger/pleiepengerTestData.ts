import { Pleiepengerettighet } from '../../../../../models/ytelse/pleiepenger';

export const pleiepengerTestData: Pleiepengerettighet = {
    barnet: '08031815240',
    omsorgsperson: '12076632144',
    andreOmsorgsperson: '26087625779',
    restDagerFomIMorgen: 678,
    forbrukteDagerTomIDag: 622,
    pleiepengedager: 1300,
    restDagerAnvist: 678,
    perioder: [
        {
            fom: '2018-11-14',
            antallPleiepengedager: 8,
            arbeidsforhold: [
                {
                    arbeidsgiverNavn: 'Andreassen ASA',
                    arbeidsgiverKontonr: '41131607047',
                    inntektsperiode: 'Månedssats',
                    inntektForPerioden: 10856,
                    refusjonTom: '2018-09-09',
                    refusjonstype: 'Ikke refusjon',
                    arbeidsgiverOrgnr: '1234567890',
                    arbeidskategori: 'Arbeidstaker'
                }
            ],
            vedtak: [
                {
                    periode: { fom: '2017-09-29', tom: '2017-10-24' },
                    kompensasjonsgrad: 63,
                    utbetalingsgrad: 100,
                    anvistUtbetaling: '2017-05-28',
                    bruttobeløp: 120,
                    dagsats: 5,
                    pleiepengegrad: 66
                },
                {
                    periode: { fom: '2017-07-18', tom: '2017-07-26' },
                    kompensasjonsgrad: 83,
                    utbetalingsgrad: 100,
                    anvistUtbetaling: '2017-06-21',
                    bruttobeløp: 596,
                    dagsats: 27,
                    pleiepengegrad: 58
                }
            ]
        },
        {
            fom: '2018-10-16',
            antallPleiepengedager: 13,
            arbeidsforhold: [
                {
                    arbeidsgiverNavn: 'Ryan og Sønner',
                    arbeidsgiverKontonr: '54938521445',
                    inntektsperiode: 'Månedssats',
                    inntektForPerioden: 45029,
                    refusjonTom: '2018-10-11',
                    refusjonstype: 'Ikke refusjon',
                    arbeidsgiverOrgnr: '1234567890',
                    arbeidskategori: 'Arbeidstaker'
                }
            ],
            vedtak: [
                {
                    periode: { fom: '2018-12-01', tom: '2018-12-09' },
                    kompensasjonsgrad: 100,
                    utbetalingsgrad: 100,
                    anvistUtbetaling: '2018-06-29',
                    bruttobeløp: 69,
                    dagsats: 12,
                    pleiepengegrad: 92
                }
            ]
        }
    ]
};
