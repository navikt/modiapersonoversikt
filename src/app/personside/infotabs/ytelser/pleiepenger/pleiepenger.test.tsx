import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Oversikt from './Oversikt';
import TestProvider from '../../../../../test/Testprovider';
import Pleiepengerperiode from './Pleiepengerperiode';
import { pleiepengerTestData } from './pleiepengerTestData';
import { setupReactQueryMocks } from '../../../../../test/testStore';

test('Om Oversikten i pleiepengeretten matcher snapshot', () => {
    setupReactQueryMocks();
    const resultat = renderer.create(
        <TestProvider>
            <Oversikt pleiepenger={pleiepengerTestData} />
        </TestProvider>
    );

    expect(resultat.toJSON()).toMatchSnapshot();
});

test('Om Pleiepengeperiode i pleiepengeretten matcher snapshot', () => {
    setupReactQueryMocks();
    const resultat = renderer.create(
        <TestProvider>
            <Pleiepengerperiode periode={pleiepengerTestData.perioder[0]} periodeNummer={1} />
        </TestProvider>
    );

    expect(resultat.toJSON()).toMatchSnapshot();
});
