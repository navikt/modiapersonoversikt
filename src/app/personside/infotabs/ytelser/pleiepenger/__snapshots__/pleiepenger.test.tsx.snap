// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Om Oversikten i pleiepengeretten matcher snapshot 1`] = `
.c1 {
  border-radius: 0.5rem;
  background-color: #f1f0f0;
  box-shadow: inset 0 0 0 0.3rem white,inset 0 0 0 0.37rem rgba(0,0,0,0.15);
  padding: 1.25rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 dt {
  color: #645f5a;
}

.c2 dd {
  font-weight: bold;
  margin-top: 0.3rem;
}

.c2 > div {
  margin: 0.9375rem 0;
  padding-right: 1rem;
  width: 13rem;
  overflow-wrap: break-word;
}

.c4 span:not(:last-child):after {
  content: '.';
}

.c3 li:not(:first-child) {
  border-top: solid 0.0625rem #b7b1a9;
}

.c0 {
  padding: .8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 > * {
  -webkit-flex-basis: 40%;
  -ms-flex-preferred-size: 40%;
  flex-basis: 40%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h3
      className="typo-undertittel"
    >
      Om pleiepengeretten
    </h3>
    <dl
      className="c2"
    >
      <div>
        <dt
          className="typo-normal"
        >
          Fra og med
        </dt>
        <dd
          className="typo-normal"
        >
          01.01.1970
        </dd>
      </div>
      <div>
        <dt
          className="typo-normal"
        >
          Til og med
        </dt>
        <dd
          className="typo-normal"
        >
          01.01.1970
        </dd>
      </div>
      <div>
        <dt
          className="typo-normal"
        >
          Pleiepengegrad
        </dt>
        <dd
          className="typo-normal"
        >
          66%
        </dd>
      </div>
      <div>
        <dt
          className="typo-normal"
        >
          Barnet (jente)
        </dt>
        <dd
          className="typo-normal"
        >
          08031815240
        </dd>
      </div>
      <div>
        <dt
          className="typo-normal"
        >
          Annen forelder
        </dt>
        <dd
          className="typo-normal"
        >
          26087625779
        </dd>
      </div>
    </dl>
  </div>
  <div
    className="c1"
  >
    <h3
      className="typo-undertittel"
    >
      Arbeidssituasjon
    </h3>
    <ol
      aria-label="Arbeidsgivere"
      className="c3"
    >
      <li>
        <dl
          className="c2"
        >
          <div>
            <dt
              className="typo-normal"
            >
              Arbeidsgiver
            </dt>
            <dd
              className="typo-normal"
            >
              Ryan og Sønner
            </dd>
          </div>
          <div>
            <dt
              className="typo-normal"
            >
              Arbeidskategori
            </dt>
            <dd
              className="typo-normal"
            >
              Arbeidstaker
            </dd>
          </div>
          <div>
            <dt
              className="typo-normal"
            >
              Inntekstsperiode
            </dt>
            <dd
              className="typo-normal"
            >
              Månedssats
            </dd>
          </div>
          <div>
            <dt
              className="typo-normal"
            >
              Kontonummer
            </dt>
            <dd
              className="typo-normal"
            >
              <span
                className="c4"
              >
                <span>
                  5493
                </span>
                <span>
                  85
                </span>
                <span>
                  21445
                </span>
              </span>
            </dd>
          </div>
          <div>
            <dt
              className="typo-normal"
            >
              Refusjonstype
            </dt>
            <dd
              className="typo-normal"
            >
              Ikke refusjon
            </dd>
          </div>
          <div>
            <dt
              className="typo-normal"
            >
              Inntekt for perioden
            </dt>
            <dd
              className="typo-normal"
            >
              45,029.00 NOK
            </dd>
          </div>
          <div>
            <dt
              className="typo-normal"
            >
              Refusjon til dato
            </dt>
            <dd
              className="typo-normal"
            >
              01.01.1970
            </dd>
          </div>
        </dl>
      </li>
    </ol>
    <button
      className="knapp knapp--hoved"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      Vis alle arbeidsforhold
    </button>
  </div>
</div>
`;

exports[`Om Pleiepengeperiode i pleiepengeretten matcher snapshot 1`] = `
.c0 > *:first-child {
  background-color: #cce1f3;
  padding: 0.5rem 1.25rem;
}

.c0 ol {
  padding: 0;
  margin: 0;
}

.c2 {
  overflow: auto;
  box-shadow: 0 0 0 0.07rem rgba(0,0,0,0.25);
  border-radius: .25rem;
  background-color: white;
}

.c2 [role='table'] {
  text-align: left;
  width: 100%;
}

.c2 [role='table'] thead,
.c2 [role='table'] .thead[role='rowgroup'] {
  border-bottom: 0.2rem solid rgba(0,0,0,0.15);
  font-weight: bold;
}

.c2 [role='table'] [role='cell'],
.c2 [role='table'] [role='columnheader'] {
  padding: 0.7rem;
  vertical-align: middle;
}

.c2 [role='table'] [role='columnheader'] {
  padding-bottom: 0.7rem;
}

.c2 [role='table'] [role='row']:nth-child(even) {
  background-color: rgba(0,0,0,0.1);
}

.c1 {
  border-radius: 0.5rem;
  background-color: #f1f0f0;
  box-shadow: inset 0 0 0 0.3rem white,inset 0 0 0 0.37rem rgba(0,0,0,0.15);
  padding: 1.25rem;
  margin: .8rem;
}

.c1 > *:not(:last-child) {
  margin-bottom: 1rem;
}

<li
  className="c0"
>
  <h3
    className="typo-undertittel"
  >
    Periode 1 - 01.01.1970
  </h3>
  <div
    className="c1"
  >
    <h2
      className="typo-undertittel"
    >
      Anviste utbetalinger
    </h2>
    <div
      className="typo-normal c2"
    >
      <table
        role="table"
      >
        <thead
          role="rowgroup"
        >
          <tr
            role="row"
          >
            <th
              role="columnheader"
            >
              Fra og med
            </th>
            <th
              role="columnheader"
            >
              Til og med
            </th>
            <th
              role="columnheader"
            >
              Bruttobeløp
            </th>
            <th
              role="columnheader"
            >
              Anvist utbetaling
            </th>
            <th
              role="columnheader"
            >
              Dagsats
            </th>
            <th
              role="columnheader"
            >
              Pleiepengegrad
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              01.01.1970
            </td>
            <td
              role="cell"
            >
              01.01.1970
            </td>
            <td
              role="cell"
            >
              120.00 NOK
            </td>
            <td
              role="cell"
            >
              01.01.1970
            </td>
            <td
              role="cell"
            >
              5.00 NOK
            </td>
            <td
              role="cell"
            >
              66%
            </td>
          </tr>
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              01.01.1970
            </td>
            <td
              role="cell"
            >
              01.01.1970
            </td>
            <td
              role="cell"
            >
              596.00 NOK
            </td>
            <td
              role="cell"
            >
              01.01.1970
            </td>
            <td
              role="cell"
            >
              27.00 NOK
            </td>
            <td
              role="cell"
            >
              58%
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</li>
`;
